script.js

```diff
@@ -1,12 +1,155 @@
+
+    document.getElementById("root").appendChild(episodeDivContainer);
+    episodeDivContainer.appendChild(episodeTitleElement);
+    episodeDivContainer.appendChild(episodeImageElement);
+    episodeDivContainer.appendChild(episodeSummary);
+  });
 }
 
-window.onload = setup;
+function searchShow(query) {
+  const url = `https://api.tvmaze.com/search/shows?q=${query}`;
+  fetch(url)
+    .then((response) => {
+      if (response.status >= 200 && response.status < 400) {
+        return response.json();
+      } else {
+        throw "HTTP ERROR";
+      }
+    })
+    .then((jasonData) => {
+      console.log(jasonData);
+      const results = jasonData.map((element) => ({
+        name: element.show.name,
+        image: element.show.image.medium,
+        summary: element.show.summary,
+      }));
+      renderResults(results);
+      document.getElementById("errorMessage").innerHTML = "";
+    })
+    .catch((error) => {
+      document.getElementById("errorMessage").innerHTML = error;
+      renderResults([]);
+    });
+}
+
+// show the dropdown list of episodes
+function ListEpisode(allEpisodes) {
+  allEpisodes.map((episode) => {
+    let optionElement = document.createElement("option");
+    optionElement.innerHTML = `${episode.name} - S${episode.season}E${episode.number}`;
+    selectElement.appendChild(optionElement);
+  });
+}
+
+//This function will show the dorpdown list of all shows
+function ListOfShows(allShows) {
+  showList = document.getElementById("selectShow");
+  let selectElement = document.createElement("select");
+
+  selectElement.addEventListener("change", (event) => {
+    selectedShowId = event.target.value;
+    console.log(selectedShowId);
+
+    const theResultOfEpisode = // await showtheEpisodesList(selectedShowId);
+    console.log(theResultOfEpisode);
+    ListEpisode(theResultOfEpisode);
+  });
+
+  allShows.map((show) => {
+    let optionElement = document.createElement("option");
+    optionElement.setAttribute("value", show.id);
+    optionElement.innerText = `${show.name}`;
+    selectElement.appendChild(optionElement);
+  });
+  showList.appendChild(selectElement);
+}
+
+// this function is using fetch to return the eposide list
+function showtheEpisodesList(selectedShowId) {
+  const url = `https://api.tvmaze.com/shows/${selectedShowId}/episodes`;
+  fetch(url)
+    .then((response) => {
+      if (response.status >= 200 && response.status < 400) {
+        return response.json();
+      } else {
+        throw "HTTP ERROR";
+      }
+    })
+    .then((jsonData) => {
+      console.log(jsonData);
+      return jsonData;
+    })
+    .catch((error) => {
+      document.getElementById("errorMessage").innerHTML = error;
+    });
+}
+
+// this function to show the search results on the website.
+function renderResults(results) {
+  const list = document.getElementById("resultsList");
+  list.innerHTML = "";
+  results.forEach((result) => {
+    const elementDiv = document.createElement("div");
```

Comment:
you could use the makePageForEpisodes here, rather than implementing the same code again